pe/pe.py

```diff
@@ -404,6 +404,36 @@ def recurse_resources(node, parent_name):
             res.add_lines(dataless_resources)
             self.file_res.add_section(res)
 
+    # Inspired by https://github.com/CYB3RMX/Qu1cksc0pe/blob/086db196d2de289f0784ae4d8ee03f34bf10354b/Modules/winAnalyzer.py#L446
+    def check_number_of_functions(self):
+        if len(self.binary.imported_functions) + len(self.binary.exported_functions) < 20:
+            res = ResultSection("PE file contains a suspiciously low number of functions")
+            res.add_line(
+                f"There are {len(self.binary.imported_functions)} import and "
+                f"{len(self.binary.exported_functions)} export functions."
+            )
+            self.file_res.add_section(res)
+
+            # Inspired by https://github.com/viper-framework/viper-modules/blob/00ee6cd2b2ad4ed278279ca9e383e48bc23a2555/lief.py#L760
+            if len(self.binary.exported_functions) == 0:
+                ResultSection("No exported function found", parent=self.file_res)
+
+            # Inspired by https://github.com/viper-framework/viper-modules/blob/00ee6cd2b2ad4ed278279ca9e383e48bc23a2555/lief.py#L798
+            if len(self.binary.imported_functions) == 0:
+                ResultSection("No imported function found", parent=self.file_res)
+
+    # Inspired by https://github.com/viper-framework/viper-modules/blob/00ee6cd2b2ad4ed278279ca9e383e48bc23a2555/lief.py#L945
+    def check_data_directories(self):
```

Comment:
I think this could be moved into the add_resources() function. I believe that's where I do most of directory processing.

Reply:
https://github.com/CybercentreCanada/assemblyline-service-pe/pull/54/commits/ae075f4e95585c3021fbf34313e25990ad3c9411